### ISSUE TITLE
Environment-specific locks

### ISSUE THREAD

**Issue by necropolina** [github/branch-deploy#74]
Would be nice if we could toggle sticky locks for specific environments by providing the environment name after the trigger, such as  `.lock staging` & `.unlock development`. Then the lock should only affect deployments to the locked environment, so that I could lock a branch to `staging`, but still deploy to `development`. A universal lock could still be applied by omitting the environment parameter.

Additionally, it'd be nice to have an action with an environment option that checks if a lock exists. This way we could use these locks when triggering deployments via other methods such as pushes to the main branch.

Something with a pattern like:
```yaml
- name: lock-exists
  id: lock-exists
  uses: github/lock-exists
  with:
    environment: development

- name: deploy
  if: ${{ steps.lock-exists.outputs.exists == 'false' }}
  # deployment script, etc
```

Basically, I'm looking to deploy to all environments on pushes to `main` by default, but omit the deploys to development and staging when they're locked.

**Comment by GrantBirki** [comment]
@necropolina Thank you for the suggestion! This is a great feature request. I already have a standalone Action that works together with the `github/branch-deploy` Action.

This standalone Action is called [`github/lock`](https://github.com/github/lock) and is used to claim or remove locks in a headless manner. However, I should absolutely add a `check-only: true` flag to it so that it can be used  to "check" if deployment locks exist.

---

So this issue seems like a two part request:

1. Add a feature to the `github/lock` Action so that we can simply "check" if a lock exists or not
2. Adapt the code in the `github/branch-deploy` Action so that deployments and lock can optionally be targeted to a specific environment (at the user's discretion) while leaving other envs available for deployments

**Comment by GrantBirki** [comment]
@necropolina You can now check if a lock exists with this latest release -> https://github.com/github/lock/releases/tag/v1.1.0

**Comment by necropolina** [comment]
Looks like solid solutions to me! Thanks for hopping on this so quickly!!!

**Comment by puilp0502** [comment]
Hi, it looks like the request 1 was already fulfilled, but no. 2 still missing.  
Are you planning to add this feature to this action? Blocking deployments to the development environment while prod deployment is in effect seems like too much of an hassle.

**Comment by GrantBirki** [comment]
@puilp0502 it is absolutely something we still plan on adding. It will require a decent refactor of code so something I plan on doing at some point when time is available

**Comment by GrantBirki** [comment]
@necropolina + @puilp0502 You will be very exited to know that I have begun work on this feature and it has already come along nicely. Stay tuned! 📺
